### DISCUSSION TITLE
Does Data Provider use server side search or client side?

### DISCUSSION THREAD

**Discussion by bamboowonder** [Budibase/budibase#5353] [Help] [answered]
I'm asking because we have tables with many millions of rows, and I need to ensure that table pagination is not trying to do it on the client-side.
Where is the distinction between LIMIT on client versus server side?
Thanks a bunch

**Comment by melohagan** [comment, marked as answer]
Hi @bamboowonder 

Both the internal DB and SQL queries, pagination and limits are server-side. 

In addition, REST queries can make use of pagination if setup correctly: https://docs.budibase.com/docs/rest-pagination
